src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -0,0 +1,281 @@
+        filter.hide(TokenTypes.SINGLE_LINE_COMMENT);
+        filter.hide(TokenTypes.BLOCK_COMMENT_BEGIN);
+
+        final GeneratedJavaRecognizer parser =
+            new GeneratedJavaRecognizer(filter);
+        parser.setFilename(contents.getFileName());
+        parser.setASTNodeClass(DetailAST.class.getName());
+        try {
+            parser.compilationUnit();
+        }
+        catch (RecognitionException | TokenStreamException ex) {
+            final String exceptionMsg = String.format(Locale.ROOT,
+                "%s occurred during the analysis of file %s.",
+                ex.getClass().getSimpleName(), contents.getFileName());
+            throw new CheckstyleException(exceptionMsg, ex);
+        }
+
+        return (DetailAST) parser.getAST();
+    }
+
+    /**
+     * Parses Java source file.
+     * @param file the file to parse
+     * @param withComments {@link Options#WITH_COMMENTS} to include comment nodes to the AST
+     * @return DetailAST tree
+     * @throws IOException if the file could not be read
+     * @throws CheckstyleException if the file is not a valid Java source file
+     */
+    public static DetailAST parseFile(File file, Options withComments)
+            throws IOException, CheckstyleException {
+        final FileText text = new FileText(file.getAbsoluteFile(),
+            System.getProperty("file.encoding", StandardCharsets.UTF_8.name()));
+        return parseFileText(text, withComments);
+    }
+
+    /**
+     * Parses Java source file. Result AST does not contains comment nodes.
+     * @param file the file to parse
+     * @return DetailAST tree
+     * @throws IOException if the file could not be read
+     * @throws CheckstyleException if the file is not a valid Java source file
+     */
+    public static DetailAST parseFile(File file)
+            throws IOException, CheckstyleException {
+        return parseFile(file, Options.WITHOUT_COMMENTS);
+    }
+
+    /**
+     * Parses Java source file. Result AST contains comment nodes.
+     * @param file the file to parse.
+     * @return DetailAST tree
+     * @throws IOException if the file could not be read
+     * @throws CheckstyleException if the file is not a valid Java source file
+     */
+    public static DetailAST parseFileWithComments(File file)
+            throws IOException, CheckstyleException {
+        return parseFile(file, Options.WITH_COMMENTS);
+    }
+
+    /**
+     * Parse a text and return the parse tree.
+     * @param text the text to parse
+     * @param withComments {@link Options#WITH_COMMENTS} to include comment nodes to the AST
+     * @return the root node of the parse tree
+     * @throws CheckstyleException if the text is not a valid Java source
+     */
+    public static DetailAST parseFileText(FileText text, Options withComments)
+            throws CheckstyleException {
+        final FileContents contents = new FileContents(text);
+        DetailAST ast = parse(contents);
```

Comment:
I'm not sure why we have to create the `FileContents` when the parse really cares about the stuff in the `FileText`.
I think we should change `parse` to take `FileText` instead. We seem to always have `FileText` in these calls.

Reply:
This why:
```
lexer.setCommentListener(contents);
```
The FileContents is FileText + CommentListener.